src/devTools.js

```diff
@@ -136,6 +139,7 @@ function liftReducer(reducer, initialState) {
       if (currentStateIndex === stagedActions.length - 1) {
         currentStateIndex++;
       }
+      liftedAction.action.devTimestamp = Date.now();
```

Comment:
Yeah, that's cleaner. The way I currently have it implemented still attaches the timestamp, but doing so in the reducer makes less sense.

In the monitor I'm working on, I basically ignore the idea of toggling actions, so my thought was that I would just use the stagedActions array and grab the timestamp for the action matching the current state index (as well as the next action if not the last). Haven't had time to try that out yet, so maybe it wouldn't work.